Public/Src/Cache/ContentStore/Library/Stores/FileSystemContentStoreInternal.cs

```diff
@@ -2215,9 +2213,7 @@ public async Task<EvictResult> EvictAsync(Context context, ContentHashWithLastAc
                 // operations in the in-memory representation of the cache.
                 if (_settings.UseEmptyFileHashShortcut && contentHashWithPath.Hash.IsEmptyHash())
                 {
-                    FileSystem.CreateDirectory(contentHashWithPath.Path.Parent);
-                    using (await FileSystem.OpenAsync(contentHashWithPath.Path, FileAccess.Write, FileMode.Create, FileShare.None, FileOptions.None, 1).ConfigureAwait(false)) { }
-
+                    await FileSystem.CreateEmptyFileAsync(contentHashWithPath.Path);

```

Comment:
>CreateEmptyFileAsync [](start = 37, length = 20)

Will this be called anywhere else? If not, then let's revert.

Reply:
I actually really prefer to have higher level methods like this that explain immediately the purpose of the code. So I actually would encourage us to have more code like that even when it is used only once.

Reply:
I actually really prefer to have higher level methods like this that explain immediately the purpose of the code. So I actually would encourage us to have more code like that even when it is used only once.

---
In reply to: [280620653](https://github.com/Microsoft/BuildXL/pull/233#discussion_r280620653) [](ancestors = 280620653)